src/test/java/ch/wisv/areafiftylan/EventRestIntegrationTest.java

```diff
@@ -0,0 +1,769 @@
+    protected Event testEvent;
+
+    private Team team1;
+    private Team team2;
+
+    private User teamCaptain;
+
+    @Autowired
+    EventRepository eventRepository;
+
+    @Autowired
+    TeamRepository teamRepository;
+
+    @Autowired
+    UserRepository userRepository;
+
+    @Before
+    public void setupEventTests() {
+        testEvent = new Event("Test Event", 2, 5);
+        testEvent.setOpenForRegistration(true);
+        testEvent = eventRepository.save(testEvent);
+
+        teamCaptain = userRepository
+                .save(new User("captain", new BCryptPasswordEncoder().encode("password"), "captain@team.com"));
+
+        team1 = teamRepository.save(new Team("team1", teamCaptain));
+        team2 = new Team("team2", teamCaptain);
+        team2.addMember(user);
+        team2 = teamRepository.save(team2);
+    }
+
+    @After
+    public void cleanupEventTests() {
+        eventRepository.deleteAll();
+        teamRepository.deleteAll();
+    }
+
+    //region Test Add Events
+
+    @Test
+    public void testAddEventAsAnon() {
+        Map<String, String> eventDTO = new HashMap<>();
+        eventDTO.put("name", "event");
+        eventDTO.put("teamLimit", "10");
+        eventDTO.put("teamSize", "2");
+        //@formatter:off
+        given().
+        when().
+            content(eventDTO).contentType(ContentType.JSON).
+            post(EVENTS_ENDPOINT).
+        then().
+            statusCode(HttpStatus.SC_FORBIDDEN);
+        //@formatter:on
+    }
+
+    @Test
+    public void testAddEventAsUser() {
+        Map<String, String> eventDTO = new HashMap<>();
+        eventDTO.put("name", "event");
+        eventDTO.put("teamLimit", "10");
+        eventDTO.put("teamSize", "2");
+
+        SessionData login = login("user");
+
+        //@formatter:off
+        given().
+            filter(sessionFilter).
+            header(login.getCsrfHeader()).
+        when().
+            content(eventDTO).contentType(ContentType.JSON).
+            post(EVENTS_ENDPOINT).
+        then().
+            statusCode(HttpStatus.SC_FORBIDDEN);
+        //@formatter:on
+    }
+
+    @Test
+    public void testAddEventAsAdmin() {
+        Map<String, String> eventDTO = new HashMap<>();
+        eventDTO.put("name", "event");
+        eventDTO.put("teamLimit", "10");
+        eventDTO.put("teamSize", "2");
+
+        SessionData login = login("admin");
+
+        //@formatter:off
+        given().
+            filter(sessionFilter).
+            header(login.getCsrfHeader()).
+        when().
+            content(eventDTO).contentType(ContentType.JSON).
+            post(EVENTS_ENDPOINT).
+        then().
+            statusCode(HttpStatus.SC_CREATED).
+            header("location", containsString("/events/")).
+            body("object.eventName", is("event"));
+        //@formatter:on
+    }
+
+    private void addEvent(String username, String name, String teamLimit, String teamSize) {
```

Comment:
I'd group the Tests together as well as the private helper methods. This makes for an easier search in code.